File: scala_zio_bench/src/main/scala/com/example/helloworld/GreeterImpl.scala
package io.grpc.examples.helloworld

import io.grpc.Status
import helloworld.helloworld.ZioHelloworld.ZGreeter
import helloworld.helloworld.{HelloReply, HelloRequest}
import zio.{ZEnv, ZIO}


object GreeterImpl extends ZGreeter[ZEnv, Any] {
  override def sayHello(request: HelloRequest): ZIO[ZEnv, Status, HelloReply] =
    ZIO.succeed(HelloReply(request.request))
}

File: java_wire_plain_bench/src/main/kotlin/helloworld/GreeterImpl.kt
package helloworld


import io.grpc.stub.StreamObserver

class GreeterImpl : GreeterWireGrpc.GreeterImplBase() {
  override fun SayHello(request: HelloRequest, response: StreamObserver<HelloReply>) {

    response.onNext(HelloReply(
      response = request.request
    ))
    response.onCompleted()

  }
}
